### ISSUE TITLE
[BUG] Mailbox isn't owned by user, can't save messages

### ISSUE THREAD

**Issue by shruuub** [docker-mailserver/docker-mailserver#3221]
### Miscellaneous first checks

- [X] I checked that all ports are open and not blocked by my ISP / hosting provider.
- [X] I know that SSL errors are likely the result of a wrong setup on the user side and not caused by DMS itself. I'm confident my setup is correct.

### Affected Component(s)

Sending messages works fine, however, when a email is sent to the user, after the mailserver completes all the checks, it says (stripped):cannot update mailbox /var/mail/user

### What happened and when does this occur?

```
Apr  5 08:07:56 mail postfix/qmgr[1162]: 46CA45004FFD: from=<hello@example.com>, size=2151, nrcpt=1 (queue active)
Apr  5 08:07:56 mail postfix/trivial-rewrite[1865]: warning: do not list domain mail.example.com in BOTH mydestination and virtual_mailbox_domains
Apr  5 08:07:56 mail postfix/local[2876]: warning: specify "strict_mailbox_ownership = no" to ignore mailbox ownership mismatch
Apr  5 08:07:56 mail postfix/local[2876]: 46CA45004FFD: to=<mail@mail.example.com>, relay=local, delay=357, delays=357/0.01/0/0.02, dsn=4.2.0, status=deferred (cannot update mailbox /var/mail/mail for user mail. destination /var/mail/mail is not owned by recipient)
```

Sending a mail from the real domain to the subdomain dms is hosted on to the user mail. The same occures for every other mailprovider though. Everything works, but the email isn't saved, supposedly because of permission errors. Probably an easy fix, but I'm not sure how, and the interwebs didn't give anything useful as well.

### What did you expect to happen?

Messages getting saved on the server and showing up in my IMAP inbox folder.

### How do we replicate the issue?

1.Run a mailserver (on a subdomain)
2. Add an user
3.Send an email to said user from an external account.

### DMS version

v.11.3.0

### What operating system is DMS running on?

Linux

### Which operating system version?

Debian 11

### What instruction set architecture is DMS running on?

AMD64 / x86_64

### What container orchestration tool are you using?

Docker

### docker-compose.yml

```yml
services:
  mailserver:
    image: docker.io/mailserver/docker-mailserver:11.3.0
    container_name: mailserver
    # Provide the FQDN of your mail server here (Your DNS MX record should point to this value)
    hostname: mail.example.com
    domainname: mail.example.com
    ports:
      - "25:25"
      - "587:587"
      - "465:465"
      - "993:993"
    volumes:
      - ./docker-data/dms/mail-data/:/var/mail/
      - ./docker-data/dms/mail-state/:/var/mail-state/
      - ./docker-data/dms/mail-logs/:/var/log/mail/
      - ./docker-data/dms/config/:/tmp/docker-mailserver/
      - /etc/localtime:/etc/localtime:ro
      - ./docker-data/certbot/certs/:/etc/letsencrypt
    environment:
      - ENABLE_FAIL2BAN=1
      # Using letsencrypt for SSL/TLS certificates
      - SSL_TYPE=letsencrypt
      # Allow sending emails from other docker containers
      # Beware creating an Open Relay: https://docker-mailserver.github.io/docker-mailserver/latest/config/environment/#permit_docker
      - PERMIT_DOCKER=network
      # You may want to enable this: https://docker-mailserver.github.io/docker-mailserver/latest/config/environment/#spoof_protection
      # See step 8 below, which demonstrates setup with enabled/disabled SPOOF_PROTECTION:
      - SPOOF_PROTECTION=0
    cap_add:
      - NET_ADMIN # For Fail2Ban to work
    restart: always
  ipv6nat:
    image: robbertkl/ipv6nat
    restart: always
    network_mode: "host"
    cap_add:
      - NET_ADMIN
      - SYS_MODULE
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /lib/modules:/lib/modules:ro


networks:
  default:
    driver: bridge
    enable_ipv6: true
    ipam:
      driver: default
      config:
        - subnet: fd00:0123:4567::/48
          gateway: fd00:0123:4567::1
```

### What level of experience do you have with Docker and mail servers?

- [ ] I am inexperienced with docker
- [X] I am rather experienced with docker
- [X] I am inexperienced with mail servers
- [ ] I am rather experienced with mail servers
- [X] I am uncomfortable with the CLI
- [ ] I am rather comfortable with the CLI

### Code of conduct

- [X] I have read this project's [Code of Conduct](https://github.com/docker-mailserver/docker-mailserver/blob/master/CODE_OF_CONDUCT.md) and I agree
- [X] I have read the [README](https://github.com/docker-mailserver/docker-mailserver/blob/master/README.md) and the [documentation](https://docker-mailserver.github.io/docker-mailserver/edge/) and I searched the [issue tracker](https://github.com/docker-mailserver/docker-mailserver/issues?q=is%3Aissue) but could not find a solution

**Comment by georglauterbach** [comment]
> Sending a mail from the real domain to the subdomain dms is hosted on to the user mail. The same occures for every other mailprovider though.

Now, this is a first for me. What are you actually trying to do? Are you sending an email from `user@example.com` to `user@mail.example.com`? And if so, why? To me, no offense, this looks either like a _very_ special use case or a misconception (that we see often) whereby users think because the mail server runs on a subdomain, the mail address is for the subdomain as well - please clarify.

---

Please remove the following line from your `docker-compose.yaml` file:

```yaml
domainname: mail.example.com
```

**Comment by georglauterbach** [comment]
What is the output of `setup email list` when run inside the container? (anonymize of needed)

**Comment by shruuub** [comment]
> > Sending a mail from the real domain to the subdomain dms is hosted on to the user mail. The same occures for every other mailprovider though.
> 
> Now, this is a first for me. What are you actually trying to do? Are you sending an email from `user@example.com` to `user@mail.example.com`? And if so, why? To me, no offense, this looks either like a _very_ special use case or a misconception (that we see often) whereby users think because the mail server runs on a subdomain, the mail address is for the subdomain as well - please clarify.
> 
> Please remove the following line from your `docker-compose.yaml` file:
> 
> ```yaml
> domainname: mail.example.com
> ```

My bad, forgot to clarify here. I just use mail.example.com right now, because example.com is already hosted at a 3rd party mail provider. I want to switch sooner or later, but I thought, for testing purposes, I might as well send it from a subdomain, which just happened to be mail.example.com. Is the domainname line deprecated or is it just so the mailserver would send mails from example.com? Will post the output in a second comment, only have the ssh key on another device.

**Comment by shruuub** [comment]
> What is the output of `setup email list` when run inside the container? (anonymize of needed)

`* mail@mail.example.com ( 1.0K / ~ ) [0%]`

**Comment by georglauterbach** [comment]
> > Please remove the following line from your `docker-compose.yaml` file:
> > ```yaml
> > domainname: mail.example.com
> > ```
> 
> [...] Is the domainname line deprecated or is it just so the mailserver would send mails from example.com? Will post the output in a second comment, only have the ssh key on another device.

`docker-compose.yml`'s `domainname:` field is for NIS and not for DNS IIRC. You don't need it and it potentially messes with our configs. So removing it is the right course of action.

---

> My bad, forgot to clarify here. I just use mail.example.com right now, because example.com is already hosted at a 3rd party mail provider. I want to switch sooner or later, but I thought, for testing purposes, I might as well send it from a subdomain, which just happened to be mail.example.com. 

Ah, now I see. I thought this was all on the same host 😆 I'd simplify your setup first. Use the default for `PERMIT_DOCKER`. Before that, can you create another address that does not start with `mail`, something like `test@mail.example.com` and try sending the mail again. It could be the case that the `mail@` parts messes with our script (or Postfix) somewhere...

**Comment by shruuub** [comment]
> > > Please remove the following line from your `docker-compose.yaml` file:
> > > ```yaml
> > > domainname: mail.example.com
> > > ```
> > 
> > 
> > [...] Is the domainname line deprecated or is it just so the mailserver would send mails from example.com? Will post the output in a second comment, only have the ssh key on another device.
> 
> `docker-compose.yml`'s `domainname:` field is for NIS and not for DNS IIRC. You don't need it and it potentially messes with our configs. So removing it is the right course of action.
> 
> > My bad, forgot to clarify here. I just use mail.example.com right now, because example.com is already hosted at a 3rd party mail provider. I want to switch sooner or later, but I thought, for testing purposes, I might as well send it from a subdomain, which just happened to be mail.example.com.
> 
> Ah, now I see. I thought this was all on the same host 😆 I'd simplify your setup first. Use the default for `PERMIT_DOCKER`. Before that, can you create another address that does not start with `mail`, something like `test@mail.example.com` and try sending the mail again. It could be the case that the `mail@` parts messes with our script (or Postfix) somewhere... If none of the above works, try changing the tag from `v11.3.0` to `:edge` and see if that helps.

Nothing really worked, or at least, if it did, I've got a new error.
```
mailserver            | Apr  5 22:33:40 mail postfix/trivial-rewrite[554]: warning: do not list domain mail.example.com in BOTH mydestination and virtual_mailbox_domains
mailserver            | Apr  5 22:33:41 mail policyd-spf[721]: prepend Received-SPF: Pass (mailfrom) identity=mailfrom; client-ip=0.0.0.0; helo=mail.example.com; envelope-from=example@example.com receiver=<UNKNOWN>
mailserver            | Apr  5 22:33:41 mail postfix/smtpd[716]: NOQUEUE: reject: RCPT from mail.example.com[0.0.0.0]: 550 5.1.1 <test@mail.example.com>: Recipient address rejected: User unknown in local recipient table; from=<example@example.com> to=<test@mail.example.com> proto=ESMTP helo=<mail.example.com>
mailserver            | Apr  5 22:33:41 mail postfix/smtpd[716]: disconnect from mail.example.com[0.0.0.0] ehlo=2 starttls=1 mail=1 rcpt=0/1 rset=1 quit=1 commands=6/7
```
Not sure if this is a SPF error, but yeah, anonymized hostnames and ip adresses. For the record, this is my SPF record.
```
v=spf1 include:spf.example.com ~all
```
And then the actual record is
```
v=spf1 ip4:0.0.0.0
```
actually, looking at it a second time, that does in fact seem wrong, but I'm a bit to tired to re-do it right now. Will update my comment if I re-do the record.

**Comment by georglauterbach** [comment]
Just making sure here: did you add the address `test@mail.example.com`? Asking because you said you're tired.

Then please remove all DKIM/SPF/DMARC records so we can try the simplest setup possible.

**Comment by polarathene** [comment]
IIRC the `dns.sh` helper script supports bare domain or an FQDN which truncates the first DNS label. The latter was presumably done as a convenience implicitly, but I think affects Postfix recognizing itself (might only be internally).

Thus `@mail.example.com` => `@example.com` and `@no.really.example.com` => `@really.example.com` :man_shrugging: 

https://github.com/docker-mailserver/docker-mailserver/blob/637d27efc72e11b50da69b262943d3ca29aea14f/target/scripts/helpers/dns.sh#L49-L59

That would affect anything using `DOMAINNAME` variable internally, such as `ssl.sh` helper I think for looking up certs?

I do recall a recent issue with the issue I'm describing, and I detailed it much better, I don't have time to review it and relay here, but it may be helpful: https://github.com/docker-mailserver/docker-mailserver/issues/3152#issuecomment-1455014405

Apologies, tired :sweat_smile:

**Comment by georglauterbach** [comment]
Then this is also kind of related to #3197.

**Comment by shruuub** [comment]
> Just making sure here: did you add the address `test@mail.example.com`? Asking because you said you're tired.
> 
> Then please remove all DKIM/SPF/DMARC records so we can try the simplest setup possible.

Sorry for the late response, but yes, I'm pretty sure, since a) the user shows up in setup.sh, and b) login on mailclients works. Will remove the records and report back.

For polarathene's comment, the second part might actually be my problem, since my mailserver also trys to connect to the mailserver at the root domain like in #3152 .

**Comment by shruuub** [comment]
Interestingly, it also seems that all users except mail do theoretically exist (IMAP login works, not sure bout POP) and thus they are shown in email list, but postfix doesn't actually recognize them, restarting changes nothing as well. Might later try a domain, not sure if I'll find one that isn't a subdomain though (like eu.org etc.)

**Comment by polarathene** [comment]
> IMAP login works, not sure bout POP

These are for retrieving mail, that is via Dovecot, different software in the container. Receiving and sending is via Postfix.

Our docs are getting revised, [here is a preview](https://pullrequest-3219--dms-doc-previews.netlify.app/introduction/#overview) of a relevant section for you to get a better understanding of these differences.

> eu.org

- `user@example.eu.org` (email address) => `mail.example.eu.org` (MX)
- `user@sub.example.eu.org` (email address) => `mail.sub.example.eu.org` (MX)

AFAIK, this is the expected behaviour with Postfix in our project presently. That's all you should need to account for. The issue with having a different mail host already for `example.com` just means you would need to add one more DNS label like shown `mail.sub.example.com`. Or you can try make `mail.example.com` work in this case with `OVERRIDE_HOSTNAME=mail.example.com` I think, similar to advice in docs FAQ for bare domain.

You can also refer to #3197 which talks about the issue with Postfix config change. But if you don't want to deal with any of that easiest fix atm is to rely on the current implicit behaviour with `mail.sub.example.com`, or get a new domain for `mail.other-domain.com`.

**Comment by shruuub** [comment]
I see! The PR does look good, I did sorta wonder about postfix, since I recalled it being for sending, but yeah. Might try a domain then, and will report back again.

**Comment by shruuub** [comment]
Alright, I finally got to re-doing the whole setup, and with a different domain everything worked. Thanks for the great and fast support, this project rocks! :)
